src/main/java/org/jabref/gui/commonfxcontrols/CitationKeyPatternSuggestionCell.java

```diff
@@ -191,5 +209,62 @@ private Menu createPatternsSubMenu() {
             }
             return patternsSubMenu;
         }
+
+        private Menu createCompoundPatternsSubMenu() {
+            Menu compoundPatternsSubMenu = new Menu(Localization.lang("Create compound pattern"));
+
+            Map<CitationKeyPattern.Category, List<CitationKeyPattern>> categorizedPatterns =
+                    CitationKeyPattern.getAllPatterns().stream()
+                                      .collect(Collectors.groupingBy(CitationKeyPattern::getCategory));
+
+            Map<CitationKeyPattern.Category, String> categoryNames = new LinkedHashMap<>();
+            categoryNames.put(CitationKeyPattern.Category.AUTHOR_RELATED, Localization.lang("Author related"));
+            categoryNames.put(CitationKeyPattern.Category.EDITOR_RELATED, Localization.lang("Editor related"));
+            categoryNames.put(CitationKeyPattern.Category.TITLE_RELATED, Localization.lang("Title related"));
+            categoryNames.put(CitationKeyPattern.Category.OTHER_FIELDS, Localization.lang("Other fields"));
+            categoryNames.put(CitationKeyPattern.Category.BIBENTRY_FIELDS, Localization.lang("Entry fields"));
+
+            for (Map.Entry<CitationKeyPattern.Category, String> entry : categoryNames.entrySet()) {
+                CitationKeyPattern.Category category = entry.getKey();
+                String categoryName = entry.getValue();
+
+                Menu categoryMenu = new Menu(categoryName);
+                List<CitationKeyPattern> patterns = categorizedPatterns.getOrDefault(category, List.of());
+
+                for (CitationKeyPattern pattern : patterns) {
+                    MenuItem menuItem = new MenuItem(pattern.stringRepresentation());
+                    menuItem.setOnAction(event -> {
+                        setText(pattern.stringRepresentation());
+                        positionCaret(pattern.stringRepresentation().length());
+                        suggestionsList.hide();
+
+                        if (selectionCount == 0) {
+                            selectionCount++;
+                            selectedPatterns.append(pattern.stringRepresentation());
+                            setText(selectedPatterns.toString());
+                            positionCaret(selectedPatterns.length());
+                            dialogService.notify(Localization.lang("Select one more item to create a compound pattern."));
+                            if (getScene() != null) {
+                                suggestionsList.getItems().remove(ALL_PATTERNS_POSITION);
+                                double screenX = localToScreen(0, 0).getX();
+                                double screenY = localToScreen(0, 0).getY() + getHeight();
+                                suggestionsList.show(this, screenX, screenY);
+                            }
+                        } else if (selectionCount == 1) {
+                            selectedPatterns.append("_").append(pattern.stringRepresentation());
```

Comment:
Not sure how this whole things works. Somewhere, it should be explained, how `selectedPatterns` is inteded to be sued.